src/main/java/com/amazon/dlic/auth/http/saml/AuthTokenProcessorHandler.java

```diff
@@ -410,8 +410,6 @@ private String extractSubject(SamlResponse samlResponse) throws Exception {
 
         if (samlRolesSeparator != null) {
             values = splitRoles(values);
-        } else {
-            values = trimRoles(values);
         }
 
         return values.toArray(new String[values.size()]);
```

Comment:
Can `extractRoles()` return `List<String>` instead of `String[]`?

Reply:
I don't think this is required for this fix. Can you provide a reason that this is required?

Reply:
OK, if you want to fix this in a separate PR (minor optimization to avoid unnecessary conversion from `List` to array).

Reply:
Yes, separate the PR is better IMO.